contrib/check-semver.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+#
+# Checks the public API of crates, exits with non-zero if there are currently
+# changes to the public API not already committed to in the various api/*.txt
+# files.
+
+set -euo pipefail
+
+NIGHTLY=$(cat nightly-version)
+# Our docs have broken intra doc links if all features are not enabled.
+RUSTDOCFLAGS="-Z unstable-options --document-private-items --document-hidden-items --output-format=json --cap-lints=allow"
+
+if [ -n "${GITHUB_HEAD_REF+x}" ]; then
+  CURRENT_COMMIT=$GITHUB_HEAD_REF # running on CI
+else
+  CURRENT_COMMIT=$(git rev-parse HEAD)
+fi
+
+if [ -n "${GITHUB_BASE_REF+x}" ]; then
+  MASTER_COMMIT=$GITHUB_BASE_REF # running on CI
+else
+  MASTER_COMMIT=$(git rev-parse master)
+fi
+
+main() {
+    need_nightly
+
+    # on current commit
+    generate_json_files "bitcoin" "current"
+    generate_json_files "base58ck" "current"
+    generate_json_files "hashes" "current"
+    generate_json_files "units" "current"
+    generate_json_files "io" "current"
+
+    # switch to master
+    git checkout "$MASTER_COMMIT"
+
+    generate_json_files "bitcoin" "master"
+    generate_json_files "base58" "master"
+    generate_json_files "hashes" "master"
+    generate_json_files "units" "master"
+    generate_json_files "io" "master"
+
+    # Check for API semver breaks
+    run_cargo_semver_check "bitcoin"
+    run_cargo_semver_check "base58ck"
+    run_cargo_semver_check "hashes"
+    run_cargo_semver_check "units"
+    run_cargo_semver_check "io"
```

Comment:
You should generate these programmatically.

Reply:
I don't follow. Programmatically means "in a way that follows a plan or uses a particular method".
That can be anything. Can you be more specific?

Reply:
I believe he means "extract the names of the crates from the top-level Cargo.toml file".